### PR TITLE
Use server.decorate

### LINKED ISSUE #104: Use `server.decorate()` instead of assignment
See Eran's comment:
https://github.com/hapijs/hapi-auth-cookie/pull/103#discussion_r48878555

### PR DESCRIPTION
**Pull request by jaw187**
Closes #104 

Requires bump to Hapi 11 to support passing options to `server.decorate`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,11 +20,11 @@
     "joi": "7.x.x"
   },
   "peerDependencies": {
-    "hapi": ">=10.x.x"
+    "hapi": ">=11.x.x"
   },
   "devDependencies": {
     "code": "2.x.x",
-    "hapi": "10.x.x",
+    "hapi": "11.x.x",
```
Comment (hueniverse) [inline package.json]: Why not 12?

**Comment by jaw187** [comment]
@hueniverse Ok to merge?

File: lib/index.js [hunk 2]
```diff
@@ -100,15 +108,26 @@ internals.implementation = function (server, options) {
...
+
+                const reply = self.reply;
                 const session = request.auth.artifacts;
                 Hoek.assert(session, 'No active session to modify ttl on');
                 reply.state(settings.cookie, session, { ttl: msecs });
-            }
+            };
         };
 
+        return new cookieAuth();
```
Comment (hueniverse) [inline lib/index.js]: Should be `CookieAuth`

File: lib/index.js [hunk 3]
```diff
@@ -100,15 +108,26 @@ internals.implementation = function (server, options) {
...
+    };
+
+    server.decorate('request', 'cookieAuth', decoration, { apply: true });
+
+    server.ext('onPreAuth', (request, reply) => {
+
+        request.cookieAuth.reply = reply;
```
Comment (hueniverse) [inline lib/index.js]: I would add a comment that reply is only used for setting and unsetting state, and not for actual replies.

**Comment by hueniverse** [comment]
See comments. Otherwise good to go.

**Pull request merged**